File: nicodegroot/src/nl/nicasso/ql/ast/structure/Block.java
package nl.nicasso.ql.ast.structure;

import java.util.List;

import nl.nicasso.ql.ast.ASTNode;
import nl.nicasso.ql.ast.CodeLocation;
import nl.nicasso.ql.ast.statement.Statement;
import nl.nicasso.ql.visitor.StructureVisitor;

public class Block extends ASTNode {

	private final List<Statement> statements;

	public Block(List<Statement> statements, CodeLocation location) {
		super(location);
		this.statements = statements;
	}

	public List<Statement> getStatements() {
		return statements;
	}
	
	public <T> T accept(StructureVisitor<T> visitor) {
		return visitor.visit(this);
	}
	
}

File: JordyHeemskerk/QL/QL/Block.swift
//
//  Block.swift
//  QL
//
//  Created by Jordy Heemskerk on 24/02/16.
//  Copyright © 2016 Jordy Heemskerk. All rights reserved.
//

import Foundation
import SwiftParsec

struct Block: ASTNode {
    
    let statements: [Statement]
    let position: Position
    
    init(tupleInput: ([Statement], Position)) {
        let (statements, currentPosition) = tupleInput
        self.statements = statements
        self.position = currentPosition
    }
    
    func accept(visitor: ASTVisitor) {
        visitor.visit(self)
    }
    
}